modules/solid_mechanics/neml2/src/LAROMANCE6DInterpolation.cxx

```diff
@@ -0,0 +1,409 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LAROMANCE6DInterpolation.h"
+#include <fstream>
+#include <initializer_list>
+#include "neml2/misc/math.h"
+
+namespace neml2
+{
+register_NEML2_object(LAROMANCE6DInterpolation);
+
+OptionSet
+LAROMANCE6DInterpolation::expected_options()
+{
+  auto options = Model::expected_options();
+  options.doc() = "Multilinear interpolation over six dimensions (von_mises_stress, temperature, "
+                  "equivalent_plastic_strain, cell_dd_density, wall_dd_density, env_factor)";
+  // Model inputs
+  options.set<VariableName>("equivalent_plastic_strain") = VariableName("state", "ep");
+  options.set<VariableName>("von_mises_stress") = VariableName("state", "s");
+
+  options.set<VariableName>("cell_dd_density") = VariableName("old_state", "cell_dd");
+  options.set<VariableName>("wall_dd_density") = VariableName("old_state", "wall_dd");
+
+  options.set<VariableName>("temperature") = VariableName("forces", "T");
+  options.set<VariableName>("env_factor") = VariableName("forces", "env_fac");
+  // Model Outputs
+  options.set_output("output_rate");
+  // JSON
+  options.set<std::string>("model_file_name");
+  options.set<std::string>("model_file_variable_name");
+  // jit does not currently work with this
+  options.set<bool>("jit") = false; // false;
+  options.set("jit").suppressed() = true;
+  return options;
+}
+
+LAROMANCE6DInterpolation::LAROMANCE6DInterpolation(const OptionSet & options)
+  : Model(options),
+    _s(declare_input_variable<Scalar>("von_mises_stress")),
+    _T(declare_input_variable<Scalar>("temperature")),
```

Comment:
I'd refrain from using variables consisting of a single capital letter with an underscore prefix. According to the C++ standard, those are reserved for internal purposes, and we have run into issues with this before.